### PR TITLE
Reduce tests submitted to lab-cip

### PR DESCRIPTION
**Pull request by patersonc**
Marking this as draft as I'm not sure if the '*' wildcard can be used for the branchname.

### CODE REVIEW

**Review by aliceinwire — APPROVED** [review]

**Comment by patersonc** [comment]
Removing staging-skip label to see if this actually works...

**Review by gctucker — COMMENTED** [review]
It's not easy to test this on staging since none of these trees is typically built.  We can however test this easily manually by downloading the meta-data for some of these trees and running `kci_test list_jobs`.

For example (to be run in separate directories):
```
wget https://storage.kernelci.org/next/master/next-20211124/arm64/defconfig/gcc-10/{bmeta,steps,artifacts}.json
wget https://storage.kernelci.org/mainline/master/v5.16-rc2/arm64/defconfig/gcc-10/{bmeta,steps,artifacts}.json
wget https://storage.kernelci.org/stable/linux-5.10.y/v5.10.81/arm64/defconfig/gcc-10/{bmeta,steps,artifacts}.json
wget https://storage.kernelci.org/stable/linux-5.14.y/v5.14.21/arm64/defconfig/gcc-10/{bmeta,steps,artifacts}.json
wget https://storage.kernelci.org/cip/linux-5.10.y-cip/v5.10.8-8301-g3d6168cb89f6/arm64/defconfig/gcc-10/{bmeta,steps,artifacts}.json
```
Then to see which test would be run in `lab-cip` for each of these kernel builds:
```
$ ./kci_test list_jobs --lab-config=lab-cip --install-path=data/arm64-stable-5.10 | head -4
r8a774a1-hihope-rzg2m-ex baseline-nfs
r8a774a1-hihope-rzg2m-ex baseline
r8a774c0-ek874 baseline-cip-nfs
qemu_arm64-virt-gicv3 baseline-qemu-docker
$ ./kci_test list_jobs --lab-config=lab-cip --install-path=data/arm64-stable-5.14 | head -4
$ ./kci_test list_jobs --lab-config=lab-cip --install-path=data/arm64-mainline | head -4
qemu_arm64-virt-gicv3-uefi baseline_qemu
qemu_arm64-virt-gicv3 smc
r8a774a1-hihope-rzg2m-ex baseline
qemu_arm64-virt-gicv2-uefi baseline_qemu
$ ./kci_test list_jobs --lab-config=lab-cip --install-path=data/arm64-cip-5.10 | head -4
r8a774a1-hihope-rzg2m-ex ltp-pty
r8a774a1-hihope-rzg2m-ex ltp-fcntl-locktests
zynqmp-zcu102 baseline
r8a774a1-hihope-rzg2m-ex ltp-mm
```
and so on.  As one can see, the list of jobs gets generated for all these builds except stable 5.14 which isn't in the combination filter.

File: config/core/lab-configs.yaml [hunk 1]
```diff
@@ -31,6 +31,24 @@ labs:
             - ltp
             - preempt-rt
             - smc
+          tree:
+            - cip
+            - mainline
+            - next
+            - stable
+            - stable-rc
```
[the remark this answers is not in the document]
Comment (patersonc) [reply config/core/lab-configs.yaml:39]: Originally I was hoping to have the passlist filter for the trees where we want to test all branches, and then the combination filter for the trees where we want to be specific.

But that's obviously not possible, so will go with the suggested approaches. Thank you.

File: config/core/lab-configs.yaml [hunk 2]
```diff
@@ -31,6 +31,24 @@ labs:
...
+          values:
+            - ['cip', '*']
+            - ['mainline', '*']
+            - ['next', '*']
+            - ['stable', 'linux-4.4.y']
+            - ['stable', 'linux-4.19.y']
+            - ['stable', 'linux-5.10.y']
+            - ['stable-rc', 'linux-4.4.y']
+            - ['stable-rc', 'linux-4.19.y']
+            - ['stable-rc', 'linux-5.10.y']
```
[the remark this answers is not in the document]
Comment (patersonc) [reply config/core/lab-configs.yaml:51]: 👍

File: config/core/lab-configs.yaml [hunk 3]
```diff
@@ -31,6 +31,23 @@ labs:
             - ltp
             - preempt-rt
             - smc
+      - combination:
```
Comment (gctucker) [inline config/core/lab-configs.yaml:34]: One more thing to note, the jobs on staging.kernelci.org use the `kernelci` kernel tree.  So if you want lab-cip to keep running all the staging jobs, you'll need to add those branches too: `staging-mainline`, `staging-next`, `staging-stable`, `staging-cip` with the `kernelci` tree.  That's why other labs with a filter on trees have `kernelci` in their passlist.

Comment (gctucker) [reply config/core/lab-configs.yaml:34]: An exception to that, every day there's a regular mainline job run on staging but with a small subset of builds:
https://staging.kernelci.org/job/mainline/branch/master/

There's also a full linux-next job run once a week:
https://staging.kernelci.org/job/next/branch/master/

These jobs should match the current filters without adding the `kernelci` tree.

Comment (patersonc) [reply config/core/lab-configs.yaml:34]: Yea I was wondering about that :P

I guess the above will be enough for now. We can always re-expand if needed. Thanks for the info.

Comment (gctucker) [reply config/core/lab-configs.yaml:34]: Maybe at least add `kernelci` / `staging-cip`?

Comment (patersonc) [reply config/core/lab-configs.yaml:34]: Sure. Done.

**Review by gctucker — APPROVED** [review]
Tested OK on staging:
https://staging.kernelci.org/test/job/kernelci/branch/staging-cip/kernel/staging-cip-20211126.0/plan/baseline/
https://staging.kernelci.org/test/job/mainline/branch/master/kernel/v5.16-rc2-162-ga4849f6000e2/plan/baseline/

The other build configs don't have results from `lab-cip`, for example `staging-stable`:
https://staging.kernelci.org/test/job/kernelci/branch/staging-stable/kernel/staging-stable-20211126.0/plan/baseline/

**Comment by gctucker** [comment]
Also confirmed that the jobs now have low priority:
https://lava.ciplatform.org/scheduler/job/551234

**Pull request merged**